src/lsp-server.ts

```diff
@@ -143,7 +143,16 @@ export class LspServer {
             onEvent: this.onTsEvent.bind(this)
         });
 
-        this.tspClient.start();
+        const started = this.tspClient.start({
+            onExit: (exitCode) => {
+                this.logger.error(`tsserver process has exited with status code "${exitCode}". Stopping the server.`);
+                // Allow the log to be dispatched to the client.
+                setTimeout(() => process.exit(1));
+            }
+        });
+        if (!started) {
+            throw new Error('tsserver process has failed to start.');
```

Comment:
What happens after this is thrown? Previously TspClient would exit the process if not started, does this propagate to a handler that will exit?

Reply:
I've tested by always throwing and in that case the LSP library is returning an error to the client which makes the client close the process.